ios/NebulaNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -132,59 +135,118 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
     }
         
     private func pathUpdate(path: Network.NWPath) {
-        //TODO: we can likely be smarter here and enumerate all the interfaces and their current addresses, only rebind if things changed
-        nebula?.rebind("network change")
+        let routeDescription = collectAddresses(endpoints: path.gateways)
+        if routeDescription != cachedRouteDescription {
+            // Don't bother to rebind if we don't have any gateways
+            if routeDescription != "" {
+                nebula?.rebind("network change to: \(routeDescription); from: \(cachedRouteDescription ?? "none")")
+            }
+            cachedRouteDescription = routeDescription
+        }
+    }
+    
+    private func collectAddresses(endpoints: [Network.NWEndpoint]) -> String {
+        var str: [String] = []
+        endpoints.forEach{ endpoint in
```

Comment:
Is order guaranteed? If not, the resulting string may not match even though the addresses are all the same.

On that note - is it necessary to convert this to a string, or can we just compare the list values?

Reply:
Just below is a sort on the resulting string array, they are stored as a full string because this is passed to nebula core as a reason for the rebind and shows up in your logs.